### PR TITLE
fix(tests): use sessionstorage for testing webchannel messages

### LINKED ISSUE #3720: Failing test Firefox Desktop Sync v3 force_auth - with an unregistered email, no uid
`./node_modules/.bin/intern-runner config=tests/intern_functional_full fxaAuthRoot=https://api-accounts.stage.mozaws.net/v1 fxaContentRoot=https://accounts.stage.mozaws.net/ fxaOauthApp=https://123done-stage.dev.lcip.org/ fxaUntrustedOauthApp=https://321done-stage.dev.lcip.org/ fxaIframeOauthApp=https://123done-stage.dev.lcip.org/iframe fxaEmailRoot=http://restmail.net fxaProduction=true grep='with an unregistered email, no uid'
`

```
× firefox on any platform - Firefox Desktop Sync v3 force_auth - with an unregistered email, no uid (1.418s)
Error: #message-fxaccounts-logout should not be present
  at Command.<anonymous>  <tests/functional/lib/helpers.js:211:19>
  at runCallback  <node_modules/intern/node_modules/leadfoot/Command.js:526:31>
  at Command.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:543:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:420:9>
  at process._tickCallback  <node.js:349:13>
  at Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:542:10>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:210:12>
  at runCallback  <node_modules/intern/node_modules/leadfoot/Command.js:526:31>
  at Command.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:543:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:420:9>
  at process._tickCallback  <node.js:349:13>
  at Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:542:10>
  at Test.registerSuite.with an unregistered email, no uid [as test]  <tests/functional/sync_v3_force_auth.js:118:10>
  at <node_modules/intern/lib/Test.js:211:24>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at runCallbacks  <node_modules/intern/node_modules/dojo/Promise.ts:11:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:317:4>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:420:9>
  at process._tickCallback  <node.js:349:13>
```

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #3720

### CODE REVIEW

**Comment by vladikoff** [comment]
@shane-tomlinson seems like some firstrun is failing with this, thoughts? Feel free to adjust this if needed

File: tests/functional/fx_firstrun_v1_sign_up.js [hunk 1]
```diff
@@ -58,11 +58,12 @@ define([
           return FunctionalHelpers.fillOutSignUp(self, email, PASSWORD);
         })
 
+        .findByCssSelector('#fxa-confirm-header')
```
Comment (shane-tomlinson) [inline tests/functional/fx_firstrun_v1_sign_up.js]: @vladikoff - the problem was here, we were checking for `fxaccounts:login` before the screen transition, so it wasn't yet fired.

**Comment by shane-tomlinson** [comment]
@vladikoff - Can you file a follow on issue to remove `listenForWebChannelMessage` and updates fx_ios_v2_sign_up.js to use `FxDesktopHelpers.testIsBrowserNotifiedOfLogin` instead of `FunctionalHelpers.testIsBrowerNotified`? Otherwise looks good!

**Comment by vladikoff** [comment]
Remaining tests to check

```
× firefox on any platform - Firstrun sign_in - sign in with an already existing account (1.847s)
Error: Browser is not notified of fxaccounts:login
  at Command.<anonymous>  <tests/functional/lib/helpers.js:755:19>
  at runCallback  <node_modules/intern/node_modules/leadfoot/Command.js:526:31>
  at Command.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:543:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:419:9>
  at process._tickCallback  <node.js:348:13>
  at Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:542:10>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:753:10>
  at runCallback  <node_modules/intern/node_modules/leadfoot/Command.js:526:31>
  at Command.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:543:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:419:9>
  at process._tickCallback  <node.js:348:13>
  at Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:542:10>
  at Test.registerSuite.sign in with an already existing account [as test]  <tests/functional/fx_firstrun_v1_sign_in.js:60:10>
  at <node_modules/intern/lib/Test.js:211:24>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at runCallbacks  <node_modules/intern/node_modules/dojo/Promise.ts:11:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:317:4>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:419:9>
  at process._tickCallback  <node.js:348:13>
× firefox on any platform - Firstrun sign_in - sign in with an existing account with the `haltAfterSignIn=true` query parameter (1.841s)
Error: Browser is not notified of fxaccounts:login
  at Command.<anonymous>  <tests/functional/lib/helpers.js:755:19>
  at runCallback  <node_modules/intern/node_modules/leadfoot/Command.js:526:31>
  at Command.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:543:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:419:9>
  at process._tickCallback  <node.js:348:13>
  at Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:542:10>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:753:10>
  at runCallback  <node_modules/intern/node_modules/leadfoot/Command.js:526:31>
  at Command.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:543:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:419:9>
  at process._tickCallback  <node.js:348:13>
  at Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:542:10>
  at Test.registerSuite.sign in with an existing account with the `haltAfterSignIn=true` query parameter [as test]  <tests/functional/fx_firstrun_v1_sign_in.js:84:10>
  at <node_modules/intern/lib/Test.js:211:24>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at runCallbacks  <node_modules/intern/node_modules/dojo/Promise.ts:11:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:317:4>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:419:9>
  at process._tickCallback  <node.js:348:13>
```

```
× firefox on any platform - FxiOS v2 sign_up - sign up, verify same browser (3.219s)
Error: Browser is not notified of login
  at Command.<anonymous>  <tests/functional/lib/helpers.js:755:19>
  at runCallback  <node_modules/intern/node_modules/leadfoot/Command.js:526:31>
  at Command.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:543:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:419:9>
  at process._tickCallback  <node.js:348:13>
  at Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:542:10>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:753:10>
  at runCallback  <node_modules/intern/node_modules/leadfoot/Command.js:526:31>
  at Command.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:543:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:419:9>
  at process._tickCallback  <node.js:348:13>
  at Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:542:10>
  at Test.registerSuite.sign up, verify same browser [as test]  <tests/functional/fx_ios_v2_sign_up.js:79:10>
  at <node_modules/intern/lib/Test.js:211:24>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at runCallbacks  <node_modules/intern/node_modules/dojo/Promise.ts:11:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:317:4>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:419:9>
  at process._tickCallback  <node.js:348:13>
```

and 

```
settings/avatar - upload a profile image
settings/avatar - attempt to use webcam for avatar (profile:change event)
```

**Comment by vladikoff** [comment]
> @vladikoff - Can you file a follow on issue to remove listenForWebChannelMessage

https://github.com/mozilla/fxa-content-server/issues/3748

**Pull request merged**